imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -0,0 +1,153 @@
+class GlowsHistL0:
+    """Data structure for storing GLOWS histogram packet data.
+    Parameters
+    ----------
+    packet : tuple[list]
+        Histogram packet yielded from space_packet_parser.generate_packets
+
+    Attributes
+    ----------
+    L0_KEYS : tuple[str]
+        Data names from the packet
+    data: dict
+        The data from the packet
+
+    Methods
+    -------
+    process_packet(packet)
+        Process the given packet into the data dictionary
+
+    _convert_histogram_data(binary_hist_data)
+        Process the binary histogram data into 8-bit bins, in a list.
+    """
+
+    L0_KEYS: tuple[str] = (
+        "SHCOARSE",
+        "STARTID",
+        "ENDID",
+        "FLAGS",
+        "SWVER",
+        "SEC",
+        "SUBSEC",
+        "OFFSETSEC",
+        "OFFSETSUBSEC",
+        "GLXSEC",
+        "GLXSUBSEC",
+        "GLXOFFSEC",
+        "GLXOFFSUBSEC",
+        "SPINS",
+        "NBINS",
+        "TEMPAVG",
+        "TEMPVAR",
+        "HVAVG",
+        "HVVAR",
+        "SPAVG",
+        "SPVAR",
+        "ELAVG",
+        "ELVAR",
+        "EVENTS",
+        "HISTOGRAM_DATA",
+    )
+
+    def __repr__(self):
+        return str(self.data)
+
+    def __init__(self, packet):
+        self.data = self.process_packet(packet)
+
+    def process_packet(self, packet) -> dict:
```

Comment:
This is a great point. My one concern is, the expected list of keys does act as an additional check on the data - if one of the expected keys is missing, then the program raises an error. Is that something we would ever expect to happen? Or is that already checked by space_packet_parser?

Reply:
I went ahead and made the change, but I'm still curious about the answer to this question.

Reply:
I see. I see your point. I think that's where validation data comes to help. The way you validated with the plot. If the packet definition didn't have a key, validation data wouldn't match. But if we track that in here too, then it would add more things to track. That's my thought :)